x-pack/plugins/observability_solution/observability_ai_assistant/server/functions/context.ts

```diff
@@ -94,38 +70,27 @@ export function registerContextFunction({
           messages.filter((message) => message.message.role === MessageRole.User)
         );
 
-        const nonEmptyQueries = compact(queries);
-
-        const queriesOrUserPrompt = nonEmptyQueries.length
-          ? nonEmptyQueries
-          : compact([userMessage?.message.content]);
-
-        queriesOrUserPrompt.push(screenDescription);
-
-        const suggestions = await retrieveSuggestions({
-          client,
-          categories,
-          queries: queriesOrUserPrompt,
-        });
+        const userPrompt = userMessage?.message.content!;
+        const queries = compact([{ text: userPrompt, boost: 3 }, { text: screenDescription }]);
 
+        const suggestions = await retrieveSuggestions({ client, queries });
         if (suggestions.length === 0) {
-          return {
-            content,
-          };
+          return { content };
         }
 
         try {
           const { relevantDocuments, scores } = await scoreSuggestions({
             suggestions,
-            queries: queriesOrUserPrompt,
+            screenDescription,
+            userPrompt,
```

Comment:
@dgieselaar Per your comment we now use `queries` that includes `boost` parameter in recall but for scoring suggestions I am getting better results by telling the LLM what each query is (user prompt and screen description).

Reply:
Actually that's a great idea!